WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -335,23 +328,63 @@ private extension AddEditCouponViewModel {
             coupon.excludeSaleItems != initialCoupon.excludeSaleItems ||
             coupon.excludedProductIds != initialCoupon.excludedProductIds ||
             coupon.excludedProductCategories != initialCoupon.excludedProductCategories
-        }()
+    }
 
-        let expiryDateUpdated: Bool = {
-            guard let oldDate = initialCoupon.dateExpires, let newDate = coupon.dateExpires else {
-                return initialCoupon.dateExpires != coupon.dateExpires
-            }
-            return !oldDate.isSameDay(as: newDate)
-        }()
+    func checkExpiryDateUpdated(for coupon: Coupon) -> Bool {
+        guard let initialCoupon = self.coupon else {
+            return false
+        }
+        guard let oldDate = initialCoupon.dateExpires?.startOfDay(timezone: timezone),
```

Comment:
Would it make sense to add a comment to explain why we are trimming time here?

Reply:
Done here: bd852aeb47afc497e14442871341078561ee5698